node/src/indexer/fake.rs

```diff
@@ -479,11 +526,13 @@ pub struct FakeIndexerManager {
     ckd_request_sender: mpsc::UnboundedSender<CKDRequestFromChain>,
 
     /// Allows nodes to be disabled during tests. See `disable()`.
-    node_disabler: HashMap<AccountId, NodeDisabler>,
+    node_disabler: BTreeMap<Uid, NodeDisabler>,
     /// Allows nodes' indexers to be paused during tests.
-    indexer_pauser: HashMap<AccountId, IndexerPauser>,
+    indexer_pauser: BTreeMap<Uid, IndexerPauser>,
     /// Allows modification of the contract.
     contract: Arc<tokio::sync::Mutex<FakeMpcContractState>>,
+
+    account_id_by_uid: Arc<Mutex<BTreeMap<Uid, AccountId>>>,
```

Comment:
This does not seem to be held over any await boundaries, so I'd recommend using the `std::sync::Mutex` over the tokio one.